src/map.cpp

```diff
@@ -5645,6 +5636,50 @@ void map::remove_field( const tripoint &p, const field_id field_to_remove )
     }
 }
 
+void map::add_splatter( const field_id type, const tripoint &where, int intensity )
```

Comment:
> Cancel on fd_null

This IMHO has to be done in `add_field()`.

> If the field type is fd_blood, splatter some blood on vehicle parts

Good suggestion, but why only blood? Any liquid (vomit, acid etc) would look nice on them. If it's a limitation of the implementation, it would be nice to extend it.